### PR TITLE
fix minor code unreachability error

### LINKED ISSUE #10772: Bug Report: some minor code unreachability error mainly in testing
### Overview of the Issue

t.Error* will report test failures but continue executing the test. 
t.Fatal* will report test failures and stop the test immediately.

**So some of the code is actually unreachable**

let me give a test example:
```go
package main

import (
	"testing"
)

// LastIndex returns the index of the last instance of x in list, or
// -1 if x is not present.
func LastIndex(list []int, x int) int {
	for i := len(list) - 1; i >= 0; i-- {
		if list[i] == x {
			return i
		}
	}
	return -1
}

func TestLastIndex(t *testing.T) {
	tests := []struct {
		list []int
		x    int
		want int
	}{
		{list: []int{1}, x: 1, want: 0},
		{list: []int{1, 1}, x: 1, want: 3}, // change want:1 to want:3, should fail
		{list: []int{2, 1}, x: 2, want: 0},
		{list: []int{1, 2, 1, 1}, x: 2, want: 1},
		{list: []int{1, 1, 1, 2, 2, 1}, x: 3, want: -1},
		{list: []int{3, 1, 2, 2, 1, 1}, x: 3, want: 111}, // change want:0 to want:111, should fail
	}
	for i, tt := range tests {
		t.Logf("index:%d", i)
		if got := LastIndex(tt.list, tt.x); got != tt.want {
			t.Errorf("LastIndex(%v, %v) = %v, want %v", tt.list, tt.x, got, tt.want)
		}
	}
}

func TestLastIndex2(t *testing.T) {
	tests := []struct {
		list []int
		x    int
		want int
	}{
		{list: []int{1}, x: 1, want: 0},
		{list: []int{1, 1}, x: 1, want: 3}, // change want:1 to want:3, should fail
		{list: []int{2, 1}, x: 2, want: 0},
		{list: []int{1, 2, 1, 1}, x: 2, want: 1},
		{list: []int{1, 1, 1, 2, 2, 1}, x: 3, want: -1},
		{list: []int{3, 1, 2, 2, 1, 1}, x: 3, want: 111}, // change want:0 to want:111, should fail
	}
	for i, tt := range tests {
		t.Logf("index:%d", i)
		if got := LastIndex(tt.list, tt.x); got != tt.want {
			t.Fatalf("LastIndex(%v, %v) = %v, want %v", tt.list, tt.x, got, tt.want) // Fatalf is equivalent to Logf followed by FailNow, so the following code will be unreachable
			continue  // unreachable
		}
	}
}

/* output:

=== RUN   TestLastIndex
    prog.go:32: index:0
    prog.go:32: index:1
    prog.go:34: LastIndex([1 1], 1) = 1, want 3
    prog.go:32: index:2
    prog.go:32: index:3
    prog.go:32: index:4
    prog.go:32: index:5
    prog.go:34: LastIndex([3 1 2 2 1 1], 3) = 0, want 111
--- FAIL: TestLastIndex (0.00s)
=== RUN   TestLastIndex2
    prog.go:53: index:0
    prog.go:53: index:1
    prog.go:55: LastIndex([1 1], 1) = 1, want 3
--- FAIL: TestLastIndex2 (0.00s)
FAIL

*/
```

As for log vitess use 

https://github.com/vitessio/vitess/blob/bf13249082bb183a657aafd9d31aac938bb75dd6/go/vt/log/log.go#L70-L71

What is actually called is

https://github.com/golang/glog/blob/9ef845f417d839250ceabbc25c1b26101e772dd7/glog.go#L1144-L1149

So same with testing.

### Reproduction Steps

No

### Binary Version

```sh
No
```

### Operating System and Environment details

```sh
No
```

### Log Fragments

_No response_

### PR DESCRIPTION
**Pull request by Abirdcfly**
Signed-off-by: Abirdcfly <fp544037857@gmail.com>

## Related Issue(s)
Fixes #10772

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by Abirdcfly** [comment]
This commit should be able to backport to an older version, but I'm not sure if it's necessary.

**Comment by systay** [comment]
Hi @Abirdcfly!

Thanks so much for your contribution. 

We are moving most of the test code to use https://github.com/stretchr/testify instead of t.Error and t.Fatal. No chance you could update these tests to use that instead of just moving the Fatals around, is there?

Instead of:

```go
tree, err := ParseNext(tokens)
if err != nil {
	t.Fatalf("[%d] ParseNext(%q) err: %q, want nil", i, input, err)
	continue
}
```

we now prefer:

```go
tree, err := ParseNext(tokens)
require.NoError(t, err)
```

File: go/vt/topo/helpers/copy.go [hunk 1]
```diff
@@ -165,7 +165,7 @@ func CopyShardReplications(ctx context.Context, fromTS, toTS *topo.Server) {
 			for _, cell := range cells {
 				sri, err := fromTS.GetShardReplication(ctx, cell, keyspace, shard)
 				if err != nil {
-					log.Fatalf("GetShardReplication(%v, %v, %v): %v", cell, keyspace, shard, err)
+					log.Errorf("GetShardReplication(%v, %v, %v): %v", cell, keyspace, shard, err)
```
Comment (systay) [inline go/vt/topo/helpers/copy.go:168]: I don't understand this change. Seems to be unrelated, right?

Comment (deepthi) [reply go/vt/topo/helpers/copy.go:168]: `Fatalf` means the func will exit, so the `continue` is unreachable. If this is really a fatal error, then we could alternatively delete the next line which says `continue`.

Comment (systay) [reply go/vt/topo/helpers/copy.go:168]: This is not test-code. I don't think it's safe to change this behaviour here.

Feel free to remove the `continue` after though - like you correctly point out, `Fatal` will exit the method anyway.

Comment (Abirdcfly) [reply go/vt/topo/helpers/copy.go]: DONE

**Review by systay — APPROVED** [review]

**Review by deepthi — APPROVED** [review]
Looks like all feedback has been addressed.
Thank you for the contribution!

**Pull request merged**